packages/fx-core/src/common/azure-hosting/utils.ts

```diff
@@ -2,23 +2,103 @@
 // Licensed under the MIT license.
 
 import { ArmTemplateResult } from "../armInterface";
+import { Inputs, TokenProvider } from "@microsoft/teamsfx-api";
+import { TokenCredentialsBase } from "@azure/ms-rest-nodeauth";
+import { PreconditionError } from "../../plugins/resource/bot/errors";
```

Comment:
It doesn't make sense that this common hosting utils import "bot/resources/messages" and "bot/errors". We can put this message into this common utils